ktor-network/nix/src/io/ktor/network/sockets/DatagramSendChannel.kt

```diff
@@ -59,28 +60,29 @@ internal class DatagramSendChannel(
             }
         }
 
-        var result = false
+        val result: Boolean
 
+        val buffer = Buffer()
         try {
-            DefaultDatagramChunkBufferPool.useInstance { buffer ->
-                element.packet.copy().readAvailable(buffer)
-
-                val bytes = element.packet.copy().readBytes()
-                val bytesWritten = sento(element, bytes)
-
-                result = when (bytesWritten) {
-                    0 -> throw IOException("Failed writing to closed socket")
-                    -1 -> {
-                        if (errno == EAGAIN) {
-                            false
-                        } else {
-                            throw PosixException.forErrno()
-                        }
+            element.packet.copy().readAvailable(buffer)
+
+            val bytes = element.packet.copy().readBytes()
+            val bytesWritten = sento(element, bytes)
+
+            result = when (bytesWritten) {
+                0 -> throw kotlinx.io.IOException("Failed writing to closed socket")
+                -1 -> {
+                    if (errno == EAGAIN) {
+                        false
+                    } else {
+                        throw PosixException.forErrno()
                     }
-                    else -> true
                 }
+
+                else -> true
             }
         } finally {
+            buffer.close()
```

Comment:
When trying to fix another issue I noticed that `buffer` may not actually be used. The packet is copied into buffer, but then never read. I guess the buffer can just be removed, right?